src/main/java/org/broadinstitute/consent/http/db/SamDAO.java

```diff
@@ -98,18 +98,26 @@ public UserStatus postRegistrationInfo(AuthUser authUser) throws Exception {
     HttpResponse response = executeRequest(request);
     String body = response.parseAsString();
     if (!response.isSuccessStatusCode()) {
-      JsonElement messageElement = JsonParser.parseString(body).getAsJsonObject().get("message");
-      String message = messageElement != null ? messageElement.getAsString() : body;
-      String errorMsg = String.format("Error posting user registration information. Email: %s. Error message: %s.",
-          authUser.getEmail(),
-          message);
+      var errorMsg = getErrorMessage(authUser, body);
       Exception e = new WebApplicationException(errorMsg, response.getStatusCode());
       logException(errorMsg, new Exception(body));
       throw e;
     }
     return new Gson().fromJson(body, UserStatus.class);
   }
 
+  public static String getErrorMessage(AuthUser authUser, String body) {
+    JsonElement messageElement = JsonParser.parseString(body).getAsJsonObject().get("message");
```

Comment:
This is where I think the source of the brittleness is. I would love to see tests that exercise the case where `body` is null, is a non-json string, or some other random value.

Reply:
Okay, if we think that we cannot rely on Sam to return a json string, then I think that makes sense. I'll add handling for those cases.